ClassHierarchyAndCollections/ClassHierarchyAndCollections/Association.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ClassHierarchyAndCollections
+{
+    /// <summary>
+    /// Defines Association and inherits from Organization and ISocialMedia.
+    /// </summary>
+    public class Association : Organization, ISocialMedia
+    {
+        public static int AssociationCount;
```

Comment:
Exposing your static counter publicly means that any caller can change the count at any time, breaking encapsulation.